pkg/model/components/cloudconfiguration.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+// CloudConfigurationOptionsBuilder prepares settings related to the backing cloud provider.
+type CloudConfigurationOptionsBuilder struct {
+	Context *OptionsContext
+}
+
+var _ loader.OptionsBuilder = &CloudConfigurationOptionsBuilder{}
+
+func (b *CloudConfigurationOptionsBuilder) BuildOptions(o interface{}) error {
+	clusterSpec := o.(*kops.ClusterSpec)
+	c := clusterSpec.CloudConfig
+	if c == nil {
+		c = &kops.CloudConfiguration{}
+		clusterSpec.CloudConfig = c
+	}
+
+	if c.ManageStorageClasses == nil {
+		c.ManageStorageClasses = fi.Bool(true)
```

Comment:
Okay. Since `CloudConfig` isn't set, you need this. I ran into nil pointer issues in my PR as well.

Reply:
this change will break backwards compatbility with openstack. We cannot now update existing clusters because it says 

spec.cloudConfig.manageStorageClasses: Forbidden: Management of storage classes and OpenStack block storage classes are both specified but disagree

Reply:
Good catch!

This doesn't really break BC per se. It only breaks when you have explicitly disabled storageclass on OpenStack.

If we don't set this, it will break for pretty much everyone else. The error message seems clear on what actions users need to take.
Alternative is to ignore this field for OpenStack and document as such, but that may be more confusing for users? Could potentially lead to misconfigurations.

But open to suggestions how to fix this. I suggest at minimum a clear "Actions needed" in the release notes.

Reply:
I did migration script for our clusters which runs `kops set cluster spec.cloudConfig.manageStorageClasses=false` after that it works fine